charmcraft/commands/store/__init__.py

```diff
@@ -722,6 +737,222 @@ def run(self, parsed_args):
         emit.message(msg.format(*args))
 
 
+class PromoteBundleCommand(BaseCommand):
+    """Promote a bundle in the Store."""
+
+    name = "promote-bundle"
+    needs_config = True
+    help_msg = "Promote a bundle to another channel in the Store"
+    overview = textwrap.dedent(
+        """
+        Promote a bundle to another channel in the Store.
+
+
+        """  # TODO (amlowe): complete this description.
+    )
+
+    def fill_parser(self, parser: "ArgumentParser") -> None:
+        """Add promote-bundle parameters to the general parser."""
+        parser.add_argument("source_channel", help="The channel from which to promote the bundle")
+        parser.add_argument("target_channel", help="The target channel for the promoted bundle")
```

Comment:
Maybe `from_channel` `to_channel` better? Or just `from`, `to`.

Reply:
That's what I originally had. I changed it because the spec used the "source" and "target" language and it was easier for me when implementing the spec. @sergiusens is there any reason to leave this with source/target language or can I call them `from_channel` and `to_channel`?